src/lib.rs

```diff
@@ -199,11 +210,66 @@ impl fmt::Display for Ulid {
     }
 }
 
+/// generator for monotonic ulids
+pub struct Generator {
+    previous: Option<Ulid>
+}
+
+/// error while trying to generate a monotonic increment in the same millisecond
+#[derive(Debug, PartialEq)]
+pub enum MonotonicError {
+    /// would overflow into the next millisecond
+    Overflow,
+}
+
+impl Generator {
+    /// create a new ulid generator for monotonically ordered ulids
+    pub fn new() -> Generator { Generator { previous: None } }
+
+    /// use Utc::now to generate a monotonically ordered ulid
+    pub fn generate(&mut self) -> Result<Ulid, MonotonicError> {
+        let now = Utc::now();
+        self.generate_from_datetime(now)
+    }
+
+    /// generate a new monotonically ordered ulid. will use previous calls timestamp to
+    /// compare and make sure that each subsequent call is monotonically increasing
+    ///
+    /// # Example
+    /// ```rust
```

Comment:
This [doctest](https://doc.rust-lang.org/rustdoc/documentation-tests.html) isn't passing.